tests/pytests/unit/utils/test_files.py

```diff
@@ -0,0 +1,157 @@
+"""
+Unit Tests for functions located in salt/utils/files.py
+"""
+
+
+import copy
+import os
+
+import pytest
+import salt.utils.files
+from tests.support.mock import patch
+
+
+def test_safe_rm():
+    with patch("os.remove") as os_remove_mock:
+        salt.utils.files.safe_rm("dummy_tgt")
+        assert os_remove_mock.called is True
+
+
+@pytest.mark.skipif(
+    os.path.exists("/tmp/no_way_this_is_a_file_nope.sh"),
+    reason="Test file exists! Skipping safe_rm_exceptions test!",
+)
+def test_safe_rm_exceptions():
+    error = False
+    try:
+        salt.utils.files.safe_rm("/tmp/no_way_this_is_a_file_nope.sh")
+    except OSError:
+        error = True
+    assert error is False
+
+
+def test_safe_walk_symlink_recursion(tmp_path):
+    tmp_path = str(tmp_path)
+    if os.stat(tmp_path).st_ino == 0:
+        pytest.xfail(reason="inodes not supported in {}".format(tmp_path))
+    os.mkdir(os.path.join(tmp_path, "fax"))
+    os.makedirs(os.path.join(tmp_path, "foo", "bar"))
+    os.symlink(os.path.join("..", ".."), os.path.join(tmp_path, "foo", "bar", "baz"))
+    os.symlink("foo", os.path.join(tmp_path, "root"))
+    expected = [
+        (os.path.join(tmp_path, "root"), ["bar"], []),
+        (os.path.join(tmp_path, "root", "bar"), ["baz"], []),
+        (os.path.join(tmp_path, "root", "bar", "baz"), ["fax", "foo", "root"], []),
+        (os.path.join(tmp_path, "root", "bar", "baz", "fax"), [], []),
+    ]
+    paths = []
+    for root, dirs, names in salt.utils.files.safe_walk(os.path.join(tmp_path, "root")):
+        paths.append((root, sorted(dirs), names))
+    if paths != expected:
+        raise AssertionError(
+            "\n".join(
+                ["got:"]
+                + [repr(p) for p in paths]
+                + ["", "expected:"]
+                + [repr(p) for p in expected]
+            )
+        )
```

Comment:
```suggestion
    assert paths == expected
```
Let pytest do the assertion diff for us, unless there's a specific reason not to?

Reply:
Sure, i can change this over. This was just a direct port without too many changes, wasn't sure what was or wasn't appropriate :)

Reply:
The less code the better, and pytest knows how to report assertion failures